internal/acctest/config/extendedproperties/extended_properties_test.go

```diff
@@ -0,0 +1,129 @@
+	}
+
+	description := func() string {
+		if extProps.GetDescription() == "" {
+			return ""
+		}
+		return fmt.Sprintf("\tdescription = \"%s\"", *extProps.Description)
+	}
+
+	multi_valued := func() string {
+		if extProps.MultiValued == nil || !extProps.GetMultiValued() {
+			return ""
+		}
+		return fmt.Sprintf("\tmulti_valued = %t\n\t\t}", extProps.GetMultiValued())
+	}
+
+	return fmt.Sprintf("%s\n\t\t%s\n\t\t%s", name(), description(), multi_valued())
+}
+
+func TestAccExtendedProperties(t *testing.T) {
+	resourceName := "myExtendedProperties"
+	initialResourceModel := client.ExtendedProperty{}
+	updatedExtendedProperty := client.NewExtendedProperty()
+	updatedExtendedProperty.Name = pointers.String("example")
+	updatedExtendedProperty.Description = pointers.String("example description")
+	updatedExtendedProperty.MultiValued = pointers.Bool(true)
+	updatedResourceModel := updatedExtendedProperty
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() { acctest.ConfigurationPreCheck(t) },
+		ProtoV6ProviderFactories: map[string]func() (tfprotov6.ProviderServer, error){
+			"pingfederate": providerserver.NewProtocol6WithError(provider.NewTestProvider()),
+		},
+		Steps: []resource.TestStep{
+			// Test empty object sent
+			{
+				Config: testAccExtendedProperties(resourceName, &initialResourceModel),
+				Check:  testAccCheckExpectedExtendedPropertiesAttributes(&initialResourceModel),
+			},
+			{
+				// Test updating some fields
+				Config: testAccExtendedProperties(resourceName, updatedResourceModel),
+				Check:  testAccCheckExpectedExtendedPropertiesAttributes(updatedResourceModel),
+			},
+			{
+				// Test importing the resource
+				Config:            testAccExtendedProperties(resourceName, updatedResourceModel),
+				ResourceName:      "pingfederate_extended_properties." + resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// Test empty object sent
+			{
+				Config: testAccExtendedProperties(resourceName, &initialResourceModel),
+				Check:  testAccCheckExpectedExtendedPropertiesAttributes(&initialResourceModel),
+			},
+		},
+	})
+}
+
+func testAccExtendedProperties(resourceName string, extendedProperty *client.ExtendedProperty) string {
+	return fmt.Sprintf(`
+resource "pingfederate_extended_properties" "%[1]s" {
+  items = [
+		%s
+  ]
+}`, resourceName,
+		extendedPropertiesHCLObj(extendedProperty),
+	)
+}
+
+// Test that the expected attributes are set on the PingFederate server
+func testAccCheckExpectedExtendedPropertiesAttributes(extendedProperty *client.ExtendedProperty) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		resourceType := "ExtendedProperties"
+		testClient := acctest.TestClient()
+		ctx := acctest.TestBasicAuthContext()
+		response, _, err := testClient.ExtendedPropertiesAPI.GetExtendedProperties(ctx).Execute()
+
+		if err != nil {
+			return err
+		}
+
+		// Verify that attributes have expected values
+		if len(response.GetItems()) > 0 {
```

Comment:
You could put all of this in one if statement rather than splitting into three.

It would also be good to have a check that the items length is zero as expected when passing in the initial resource model